src/codegen/utilities/interpreter_helpers.py

```diff
@@ -204,10 +204,19 @@ def get_interpreter_params(func):
 def get_grpc_interpreter_call_params(func, params):
     """Gets the interpreter parameters for grpc request."""
     compound_params = get_input_arguments_for_compound_params(func)
+    is_read_function = is_custom_read_function(func)
     grpc_params = []
+    has_read_array_parameter = False
     for param in params:
         if param.parameter_name not in compound_params:
-            if param.is_enum:
+            if is_read_function and "read_array" in param.parameter_name:
+                if has_read_array_parameter:
+                    continue
+                grpc_params.append(
+                    f"{camel_to_snake_case(param.size.value)}={param.parameter_name}.size"
+                )
+                has_read_array_parameter = True
```

Comment:
This was added because the parameters were getting duplicated earlier. For example, `GetDigitalPowerUpStates` method had `ChannelName` twice in metadata and this was getting reflected in the request. Now that this has been fixed as part of a PR which was merged earlier, I'll remove this and generate the parameters in the existing order.